proto/workload.proto

```diff
@@ -152,11 +157,11 @@ message Workload {
   // * Requests *to* this workload, via the tunnel_protocol, do not need to be de-tunneled.
   bool native_tunnel = 14;
 
-  // Virtual IPs defines a set of virtual IP addresses the workload can be reached at.
-  // Typically these represent Service ClusterIPs.
-  // The key is a NetworkAddress string of the format network/IP, or just an IP if network
-  // of the service desired matches the network of the workload.
-  map<string, PortList> virtual_ips = 15;
+  // Deprecated. Use `services` to map this workload to service resources.
+  map<string, PortList> virtual_ips = 15 [deprecated = true];
+
+  // The services for which this workload is an endpoint.
+  repeated NamespacedHostname services = 22;
```

Comment:
I think we need `map<NamespacedHostname, PortList>`. The port part is still important

Reply:
Doesn't the service have the ports?